ui-core/src/main/java/com/adyen/checkout/components/image/ImageLoadingExtensions.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 14/12/2022.
+ */
+
+package com.adyen.checkout.components.image
+
+import android.util.DisplayMetrics
+import android.widget.ImageView
+import androidx.annotation.DrawableRes
+import androidx.annotation.RestrictTo
+import com.adyen.checkout.components.api.LogoApi
+import com.adyen.checkout.components.ui.R
+import com.adyen.checkout.core.api.Environment
+import com.adyen.checkout.core.image.DefaultImageLoader
+import com.adyen.checkout.core.image.ImageLoader
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun ImageView.load(
+    url: String,
+    imageLoader: ImageLoader = DefaultImageLoader,
+    @DrawableRes placeholder: Int = R.drawable.ic_placeholder_image,
+    @DrawableRes errorFallback: Int = R.drawable.ic_placeholder_image,
+) {
+    setImageResource(placeholder)
+    imageLoader.load(
+        url,
+        onSuccess = { setImageBitmap(it) },
+        onError = { setImageResource(errorFallback) }
+    )
+}
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun ImageView.load(
+    environment: Environment,
+    path: String,
+    imageLoader: ImageLoader = DefaultImageLoader,
+    @DrawableRes placeholder: Int = R.drawable.ic_placeholder_image,
+    @DrawableRes errorFallback: Int = R.drawable.ic_placeholder_image,
+) {
+    load(environment.baseUrl + path, imageLoader, placeholder, errorFallback)
+}
+
+@Suppress("LongParameterList")
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun ImageView.loadLogo(
+    environment: Environment,
+    txVariant: String,
+    txSubVariant: String = "",
+    size: LogoApi.Size = LogoApi.Size.SMALL,
+    imageLoader: ImageLoader = DefaultImageLoader,
+    @DrawableRes placeholder: Int = R.drawable.ic_placeholder_image,
+    @DrawableRes errorFallback: Int = R.drawable.ic_placeholder_image,
+) {
+    val densityExtension = this.resources.displayMetrics.densityDpi.getDensityExtension()
+    val logoPath = densityExtension.buildLogoPath(txVariant, txSubVariant, size)
+    load(environment, logoPath, imageLoader, placeholder, errorFallback)
+}
+
+private fun Int.getDensityExtension(): String {
+    return when {
+        this <= DisplayMetrics.DENSITY_LOW -> "-ldpi"
+        this <= DisplayMetrics.DENSITY_MEDIUM -> "" // no extension
+        this <= DisplayMetrics.DENSITY_HIGH -> "-hdpi"
+        this <= DisplayMetrics.DENSITY_XHIGH -> "-xhdpi"
+        this <= DisplayMetrics.DENSITY_XXHIGH -> "-xxhdpi"
+        else -> "-xxxhdpi"
+    }
+}
+
+private fun String.buildLogoPath(
```

Comment:
I find this being an extension to the `densityExtension` confusing. We should have a private function that takes `densityExtension` as a parameter. That'll make line 81 more readable.

Reply:
Agree 👍